CMakeLists.txt

```diff
@@ -64,6 +64,8 @@ if (WITH_GZFILEOP)
     add_definitions(-DWITH_GZFILEOP)
 endif (WITH_GZFILEOP)
 
+option(WITH_SANITIZERS "Build with sanitizers and test/fuzz" OFF)
+option(WITH_MSAN "Build with memory sanitizer and test/fuzz" OFF)
```

Comment:
These two options makes me think the first one needs to be enabled for the second one to be allowed.
A subsequent test gives an error saying that they are incompatible however.
I think the wording of the first could be more clear. Perhaps mention what sanitizers it will enable?

Reply:
Agreed.  I will send a patch to clarify the wording, like so:
```
option(WITH_ASAN "Build with address sanitizer and all supported sanitizers other than memory sanitizer; build test/fuzz" OFF)
```

I have a remark concerning the fact that configure has a different option than cmake:
```
    [--build-fuzzers]           Build test/fuzz (disabled by default)
```
When I implemented support in cmake, there was no obvious way for cmake to build all configurations under one flag: the configure script is more flexible and allows building in the same time 4 different versions of the lib: static, shared, msan, and asan. With cmake we can only build one kind of the lib at a time, as cmake does not provide a way to change the compilation flags.

Please let me know whether you would like to have a more symmetric set of flags between cmake and configure.  I could simplify the configure script to follow the flags of cmake.

Reply:
In my opinion, the cmake options for sanitizers should not duplicate text "and test/fuzz" ... It should be separate option available when either of the two options is enabled.

Reply:
The updated patch has configure unified with cmake flags.
I will add a flag to both configure and cmake to enable build of test/fuzz.

Reply:
In 3695d0d added `configure --with-fuzzers` and `cmake -DWITH_FUZZERS=1`.